web/api/v1/api.go

```diff
@@ -602,6 +610,43 @@ type TargetDiscovery struct {
 	DroppedTargets []*DroppedTarget `json:"droppedTargets"`
 }
 
+// GlobalURLOptions contains fields used for deriving the global URL for local targets.
+type GlobalURLOptions struct {
+	ListenAddress string
+	Host          string
+	Scheme        string
+}
+
+func getGlobalURL(u *url.URL, opts GlobalURLOptions) (*url.URL, error) {
```

Comment:
As mentioned aside in https://github.com/prometheus/prometheus/pull/6794#discussion_r377932616, could you keep (re-add) the comments from the original implementation of this function?

Likely only this copy of the function will survive in the long run, so it would be good to have the comments here as well.

Reply:
Sure.